lib/cublas/linalg.jl

```diff
@@ -347,19 +347,27 @@ function LinearAlgebra.mul!(C::CuMatrix{T}, A::Diagonal{T,<:CuVector}, B::CuMatr
 end
 
 function LinearAlgebra.mul!(C::CuMatrix{T}, A::Transpose{T,<:CuMatrix}, B::Diagonal{T,<:CuVector}) where {T<:CublasFloat}
-    return dgmm!('R', CuMatrix(A), B.diag, C)
+    C .= A
+    C .*= B.diag'
```

Comment:
```suggestion
    C .*= transpose(B.diag)
```

Reply:
You don't want to conjugate the coefficients if `A` is a complex matrix.

Reply:
Hum, I think you use a trick that I don't understand because you don't transpose `A`.
Can you explain what you do exactly?

Reply:
Sure, so the transpose (which I've put in both functions) of B.diag isn't a result of A being Transpose or Adjoint, it's just to make the broadcast multiplication work on the correct dimension for B diagonal. A being Adjoint or Transpose gets realized during the broadcasted assignment to C in all cases.